docs/guides/advanced/data/4844_data.md

```diff
@@ -0,0 +1,185 @@
+made available).**
+
+This verification is done in 14 steps, divided into three parts: inside the prover, inside the state keeper, and finally
+within the L1 contract.
+
+### Inside the prover
+
+Most of this is done in either witness generator or a new circuit type 4844:
+
+| step                       | explanation                                                                                                                                                                                                                                  |
+| -------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Split data                 | Divide the data into 4096 units of 256-bit points.                                                                                                                                                                                           |
+| Create Polynomial          | Form a polynomial `p` that passes through these points. For more clarity on this, refer to [Blobspace 101][blobspace_101]                                                                                                                    |
+| Trusted setup              | Obtain the [trusted setup][trusted_setup] from Ethereum's KZG ceremony (also known as the secret point 's').                                                                                                                                 |
+| Compute Commitment         | [Calculate the commitment][compute_commitment] `(c = p(s))`.                                                                                                                                                                                 |
+| Versioned Hash:            | Generate a versioned hash (`vh = hash(c)`) which is more flexible and smaller in size than the commitment. (32 bytes vs 48)                                                                                                                  |
+| Choose a Random Point:     | Select a 'random' point `x = hash(hash(data) \|\| vh)` using the [hash of the data and the versioned hash][random_point_selection]. This is ['Fiat-Shamir' heuristic][fiat_shamir]                                                           |
+| Compute Polynomial Value:  | Calculate the value `p(x) = y`.                                                                                                                                                                                                              |
+| Prepare for Circuit Input: | Ensure that `hash(vh \|\| x \|\| y)` is provided as a [public input for the circuit][scheduler_expect_4844], based on the [Schwarz-Zippel lemma][schwarz_zippel] - if 2 polynomials are equal in a random point, there are equal everywhere. |
```

Comment:
The lemma does not state that they are equal, but bounds the probability they are not. But why to even mention this lemma here?